File: serde-generate/runtime/csharp/Serde/Bincode/BincodeSerializer.cs
// Copyright (c) Facebook, Inc. and its affiliates
// SPDX-License-Identifier: MIT OR Apache-2.0

using System;

namespace Serde.Bincode {
    public class BincodeSerializer : BinarySerializer {
        public BincodeSerializer() : base(long.MaxValue) { }
        public BincodeSerializer(byte[] buffer) : base(buffer, long.MaxValue) {}
        public BincodeSerializer(ArraySegment<byte> buffer) : base(buffer, long.MaxValue) {}

        public override void serialize_len(long value) => output.Write(value);

        public override void serialize_variant_index(int value) => output.Write(value);

        public override void sort_map_entries(int[] offsets) {
            // Not required by the format.
        }
    }
}

File: serde-generate/runtime/swift/Sources/Serde/BincodeSerializer.swift
//  Copyright (c) Facebook, Inc. and its affiliates.

import Foundation

public class BincodeSerializer: BinarySerializer {
    public init() {
        super.init(maxContainerDepth: Int.max)
    }

    override public func serialize_len(value: Int) throws {
        try serialize_u64(value: UInt64(value))
    }

    override public func serialize_variant_index(value: UInt32) throws {
        try serialize_u32(value: value)
    }

    override public func sort_map_entries(offsets _: [Int]) {
        // Not required by the format.
    }
}
